packages/flex-plugin-scripts/src/scripts/start/pluginServer.ts

```diff
@@ -0,0 +1,182 @@
+ * @private
+ */
+/* istanbul ignore next */
+export const _getLocalPlugins = () => JSON.parse(readFileSync(pluginsJsonPath)) as Plugin[];
+
+/**
+ * Generates the response headers
+ *
+ * @param port  the port the browser will be running on
+ * @private
+ */
+export const _getHeaders = (port: number) => ({
+  'Access-Control-Allow-Origin': `http://localhost:${port}`,
+  'Access-Control-Allow-Methods': 'GET',
+  'Access-Control-Allow-Headers': 'Content-Type, X-Flex-Version, X-Flex-JWE',
+  'Access-Control-Allow-Credentials': 'true',
+  'Content-Type': 'application/json',
+});
+
+/**
+ * Fetches the Plugins from Flex
+ *
+ * @param token     the JWE Token
+ * @param version   the Flex version
+ */
+/* istanbul ignore next */
+export const _getRemotePlugins = async (token: string, version: string): Promise<Plugin[]> => {
+  return new Promise((resolve, reject) => {
+    const headers = {
+      'X-Flex-JWE': token,
+    };
+    if (version) {
+      headers['X-Flex-Version'] = version;
+    }
+
+    const options = {
+      hostname: 'flex.twilio.com',
+      port: 443,
+      path: '/plugins',
+      method: 'GET',
+      headers,
+    };
+
+    https
+      .request(options, (res) => {
+        const data: Buffer[] = [];
+
+        res.on('data', (chunk) => data.push(chunk));
+        res.on('end', () => resolve(JSON.parse(Buffer.concat(data).toString())));
+      })
+      .on('error', reject)
+      .end();
+  });
+};
+
+/**
+ * Rebase plugins with local plugins
+ * @param remotePlugins   the plugins returned from Flex
+ * @private
+ */
+export const _rebasePlugins = (remotePlugins: Plugin[]) => {
+  const pkg = readPackageJson();
+
+  return _getLocalPlugins()
+    .map((plugin) => {
+      // Local main (plugin) we are running
+      if (plugin.name === pkg.name) {
+        return plugin;
+      }
+
+      // Plugin is disabled - do not load it
+      if (!plugin.enabled) {
+        return null;
+      }
+
+      // Load remote plugin from Flex
+      if (plugin.remote === true) {
+        return remotePlugins.find((p) => p.name === plugin.name);
+      }
+
+      // Backward compatibility / current way of running multiple local plugins
+      if (plugin.src) {
+        return plugin;
+      }
+
+      return null;
+    })
+    .filter(Boolean);
+};
+
+/**
+ * Basic server to fetch plugins from Flex and return to the local dev-server
+ * @param browserPort  the port of browser
+ * @private
+ */
+export const _server = (browserPort: number ) => async (req: IncomingMessage, res: ServerResponse) => {
+  const responseHeaders = _getHeaders(browserPort);
+  const { headers, method, url } = req;
+
+  if (method === 'OPTIONS') {
+    res.writeHead(200, responseHeaders);
+    return res.end();
+  }
+  if (method !== 'GET' || url !== '/plugins') {
+    res.writeHead(404, responseHeaders);
+    return res.end('Route not found');
+  }
+
+  const jweToken = headers['x-flex-jwe'] as string;
+  const flexVersion = headers['x-flex-version'] as string;
```

Comment:
move this into the `try` block below

Reply:
Why? If I move this, then I'll need to move the other 400 responses there too.

Reply:
just that line, and i still don't agree but i'll approve 😂

Reply:
the reason is you don't need that var if the if statement below returns